libraries/AP_NavEKF2/AP_NavEKF2.cpp

```diff
@@ -639,37 +639,54 @@ void NavEKF2::UpdateFilter(void)
     
     const AP_InertialSensor &ins = _ahrs->get_ins();
 
+    bool statePredictEnabled[num_cores];
     for (uint8_t i=0; i<num_cores; i++) {
         // if the previous core has only recently finished a new state prediction cycle, then
         // don't start a new cycle to allow time for fusion operations to complete if the update
         // rate is higher than 200Hz
-        bool statePredictEnabled;
         if ((i > 0) && (core[i-1].getFramesSincePredict() < 2) && (ins.get_sample_rate() > 200)) {
-            statePredictEnabled = false;
+            statePredictEnabled[i] = false;
         } else {
-            statePredictEnabled = true;
+            statePredictEnabled[i] = true;
         }
-        core[i].UpdateFilter(statePredictEnabled);
+        core[i].UpdateFilter(statePredictEnabled[i]);
```

Comment:
All cores predict states, but the state updates (but not the output predictor) for the i'th core is delayed two frames relative to the i-1 core to spread expensive fusion operations. For example if the height sensor has a very large single measurement glitch, it is important that the comparison of innovation levels between the primary and subsequent cores only be performed after both cores have had opportunity to process that measurement.